tests/integration/test_task_pool.py

```diff
@@ -114,6 +131,82 @@ async def example_flow(
     return schd
 
 
+@pytest.mark.parametrize(
+    'items, expected_task_ids, expected_bad_items, expected_warnings',
+    [
+        param(
+            ['*/foo'], ['1/foo'], [], [],
+            id="Basic"
+        ),
+        param(
+            ['1/*'],
+            ['1/foo', '1/bar'], [], [],
+            id="Name glob"
+        ),
+        param(
+            ['1/FAM'], ['1/bar'], [], [],
+            id="Family name"
+        ),
+        param(
+            ['*/foo'], ['1/foo'], [], [],
+            id="Point glob"
+        ),
+        param(
+            ['*:waiting'],
+            ['1/foo', '1/bar'], [], [],
+            id="Task state"
+        ),
+        param(
+            ['8/foo'], [], ['8/foo'], ["No active tasks matching: 8/foo"],
+            id="Task not yet spawned"
+        ),
+        param(
+            ['1/foo', '8/bar'], ['1/foo'], ['8/bar'],
+            ["No active tasks matching: 8/bar"],
+            id="Multiple items"
+        ),
+        param(
+            ['1/grogu', '*/grogu'], [], ['1/grogu', '*/grogu'],
+            ["No active tasks matching: 1/grogu",
+             "No active tasks matching: */grogu"],
+            id="No such task"
+        ),
+        param(
+            ['*'],
+            ['1/foo', '1/bar'], [], [],
+            id="No items given - get all tasks"
+        )
+    ]
+)
+async def test_filter_task_proxies_paused_start(
+    items: List[str],
+    expected_task_ids: List[str],
+    expected_bad_items: List[str],
+    expected_warnings: List[str],
+    mod_example_flow_paused_start: Scheduler,
+    caplog: pytest.LogCaptureFixture
+) -> None:
+    """Test TaskPool.filter_task_proxies().
+
+    The NOTE before EXAMPLE_FLOW_CFG above explains which tasks should be
```

Comment:
Does the note need updating? https://github.com/cylc/cylc-flow/blob/24147eaa4cda64013c1afb6753e735fd8c7f4b48/tests/integration/test_task_pool.py#L28-L29

```diff
+# NOTE: foo & bar have no parents so at start-up they are
+spawned out to the runahead limit (only when not paused at start-up).
```